example/DRCfastSDsteer.py

```diff
@@ -0,0 +1,105 @@
+# ==========================================================================
+#  AIDA Detector description implementation
+# --------------------------------------------------------------------------
+# Copyright (C) Organisation europeenne pour la Recherche nucleaire (CERN)
+# All rights reserved.
+#
+# For the licensing terms see $DD4hepINSTALL/LICENSE.
+# For the list of contributors see $DD4hepINSTALL/doc/CREDITS.
+#
+# ==========================================================================
+
+from __future__ import absolute_import, unicode_literals
+import os
+import time
+import DDG4
+from DDG4 import OutputLevel as Output
+from g4units import GeV, MeV, m
+
+def run():
+  args = DDG4.CommandLine()
+  kernel = DDG4.Kernel()
+  logger = DDG4.Logger('DRCfastSD')
+
+  kernel.loadGeometry(str("./IDEA_o1_v03/IDEA_o1_v03.xml"))
+
+  detectorDescription = kernel.detectorDescription()
+
+  DDG4.importConstants(detectorDescription, debug=False)
+
+  geant4 = DDG4.Geant4(kernel)
+  geant4.printDetectors()
+
+  # Configure UI
+  ui = geant4.setupCshUI()
+  ui.Commands = ['/run/beamOn ' + str("10"), '/ddg4/UI/terminate']
+  #
+  # Configure field
+  geant4.setupTrackingField()
+  #
+  # Configure G4 geometry setup
+  seq, act = geant4.addDetectorConstruction('Geant4DetectorGeometryConstruction/ConstructGeo')
+  #
+  # Assign sensitive detectors according to the declarations 'tracker' or 'calorimeter', etc
+  # seq, act = geant4.addDetectorConstruction('Geant4DetectorSensitivesConstruction/ConstructSD')
+  # Assign sensitive detectors in Geant4 by matching a regular expression in the detector sub-tree
+  seq, act = geant4.addDetectorConstruction('Geant4RegexSensitivesConstruction/ConstructSDRegEx')
+  act.Detector = 'DRcalo'
+  act.OutputLevel = Output.ALWAYS
+  act.Match = ['(.*)(core|clad)(.*)'] # regex is expensive, always try to avoid complex match
```

Comment:
```suggestion
  act.Match = ['(core|clad)'] # regex is expensive, always try to avoid complex match
```
Is that not enough?

Reply:
So you need the full expression, because dd4hep uses `regex_match` and not [https://en.cppreference.com/w/cpp/regex/regex_search](regex_search)?
Would regex_search be faster?

Reply:
Indeed I need the full expression, but in general I think it is already a fair trade to spend 300s to match ~ 600k volumes and save 9GB of memory (and I'm not sure how much we will gain by using `regex_search` over `regex_match`). For example, I got the following from the `lxplus9`:

```sh
ConstructSDRegEx       +++ Detector: DRcalo Handled 580482 nodes with 580482 sensitive volume type(s). Total of 237.361 seconds.
```

The most important thing from my experience is not using a list, e.g. one should avoid using
```python
act.Match = ['(.*)(core)(.*)','(.*)(clad)(.*)']
```
over
```python
act.Match = ['(.*)(core|clad)(.*)']
```
(and users can handle this).

Reply:
Using your branch and example
regex_match (with full regex)
```
ConstructSDRegEx       +++ Detector: DRcalo Handled 580482 nodes with 580482 sensitive volume type(s). Total of 108.905 seconds.
```
regex_search (with simpler regex)
```
ConstructSDRegEx       +++ Detector: DRcalo Handled 580482 nodes with 580482 sensitive volume type(s). Total of  68.064 seconds.
```

We should probably also change the verbosity level to avoid printing all matched volumes?

Reply:
Nice! And yes, we should suppress printouts.

Reply:
I'll make the changes in DD4hep, and add the RegexSD construction to ddsim.

Reply:
https://github.com/AIDASoft/DD4hep/pull/1308